packages/react-app/src/pages/details.tsx

```diff
@@ -21,8 +21,20 @@ const loader = async ({ params }: any) => {
 };
 
 const GiveawayDetailsPage = () => {
-  const userInfo = useUserInfo();
   const giveaway = useLoaderData() as Giveaway;
+  const revalidator = useRevalidator();
+  const { data: participants, refetch } = GetParticipants(giveaway._id);
+  const participationState = useRef<ParticipationState>();
+
+  const onParticipationChange = (newState: ParticipationState) => {
+    if (newState !== participationState.current) {
```

Comment:
Just a note for the future: [React query](https://tanstack.com/query/v3/) provides a better an more efficient way to manage and re-fetch queries. It also provides cache by default making the app faster, and error handling.